app/api/probe_dock/project_versions_api.rb

```diff
@@ -50,7 +50,7 @@ def current_organization
           end
 
           if params[:testId].present?
-            paginated_rel = paginated_rel.joins(test_results: :test).where('project_tests.api_id = ?', params[:testId])
+            paginated_rel = paginated_rel.joins(test_results: :test).where('project_tests.api_id = ?', params[:testId]).group('project_versions.id, projects.name')
```

Comment:
I didn't see it before, but avoid joining through `:test_results` at all costs. It will always be slow, and we might remove that table in the future (if we move the data to another DB like elastic search). You can obtain the same result by joining through `:test_descriptions`.